defaults/policies/advice/otel.rego

```diff
@@ -0,0 +1,97 @@
+package advice
+
+import rego.v1
+
+# Use pre-computed sets from jq
+attributes_set := data.attributes_set
+
+templates_set := data.templates_set
+
+namespaces_to_check_set := data.namespaces_to_check_set
+
+name_regex := "^[a-z][a-z0-9]*([._][a-z0-9]+)*$"
+
+derive_namespaces(name) := [
+concat(".", array.slice(parts, 0, i)) |
+	parts := split(name, ".")
+	count(parts) > 1 # Only derive namespaces if there are at least 2 parts
+
+	# Stop at count(parts)-1 to exclude the full attribute name
+	some i in numbers.range(1, count(parts) - 1)
+]
+
+# checks attribute has a namespace
+deny contains make_advice(key, advisory, value, message) if {
+	not contains(input.name, ".")
+	key := "missing_namespace"
+	advisory := "improvement"
+	value := input.name
+	message := "Does not have a namespace"
+}
+
+# checks attribute name format
+deny contains make_advice(key, advisory, value, message) if {
+	not regex.match(name_regex, input.name)
+	key := "invalid_format"
+	advisory := "violation"
+	value := input.name
+	message := "Does not match name formatting rules"
+}
+
+# checks attribute namespace doesn't collide with existing attributes
+deny contains make_advice(key, advisory, value, message) if {
```

Comment:
This is also something we only enforce in semconv.

Not only that - why do we need this in advice in addition to testing the repo itself for this check?

I.e. shouldn't this check *only* be done on a registry?